DoxygenLayout.xml

```diff
@@ -0,0 +1,239 @@
+<doxygenlayout version="1.0">
```

Comment:
Is this auto-generated?

Reply:
yes, auto generated and then customised

Reply:
Okay. Does the customization involve some added text? Because it doesn't seem so ideal to hide text inside an auto-generated xml. Is this for configuring the 
```
set (DOXYGEN_USE_MDFILE_AS_MAINPAGE ${CMAKE_SOURCE_DIR}/README.md)
```